qiskit_experiments/base_analysis.py

```diff
@@ -25,6 +27,36 @@ class BaseAnalysis(ABC):
     # Expected experiment data container for analysis
     __experiment_data__ = ExperimentData
 
+    def __init__(self, **options):
+        """Initialize a base analysis class
+
+        Args:
+            options: kwarg options for analysis.
+        """
+        self._options = self._default_options()
+        self.set_options(**options)
+
+    @classmethod
+    def _default_options(cls):
+        return Options()
+
+    def set_options(self, **fields):
+        """Set the analysis options.
+
+        Args:
+            fields: The fields to update the options
+        """
+        self._options.update_options(**fields)
+
+    @property
+    def options(self):
+        """Return the analysis options.
+
+        The options of an analysis class are used to provide kwarg values for
+        the :meth:`run` method.
+        """
+        return self._options
```

Comment:
There is probably a few ways to do this. One option is in the analysis class, see e.g. the work in progress here: https://github.com/Qiskit/qiskit-experiments/pull/31

Reply:
It probably makes sense to put something into the ExperimentData class for meas level. In ExperimentData when you add job/results and it pulls out circuit data it could for example store the meas_level / result_type in the metadata (if we didn't already add it as experiment metadata).